.github/workflows/unicorns.yml

```diff
@@ -0,0 +1,51 @@
+name: CI/CD pipeline
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'unicorn-pipe' #temp for debugging and testing
+defaults:
+  run:
+    shell: bash
+jobs:
+  check-unicorn:
+    name: Check if app is on unicorn track
+    runs-on: ec2-runners
+    container:
+      image: public.ecr.aws/m3u4c4h9/island-is/actions-runner-public:latest
+    timeout-minutes: 35
+
+    outputs:
+      IS_UNICORN: ${{ steps.unicorn-affected.outputs.IS_UNICORN }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get cache
+        id: get-cache
+        uses: ./.github/actions/get-cache
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          enable-cache: 'node_modules,generated-files'
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'package.json'
+
+      - name: Setup yarn
+        run: corepack enable
+
+      - name: Check unicorn affected
+        id: unicorn-affected
+        run: |
+          yarn nx show projects --base origin/main --affected
```

Comment:
```suggestion
          yarn nx show projects --base ${{ github.event.repository.default_branch }} --affected
```

Reply:
Yes, eventually - Though I'd keep it here since next step is also something we need to develop in a controlled environment